packages/kbn-securitysolution-io-ts-alerting-types/src/frequency/index.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import * as t from 'io-ts';
+
+import { RuleActionThrottle } from '../throttle';
+
+/**
+ * Action summary indicates whether we will send a summary notification about all the generate alerts or notification per individual alert
+ */
+export type RuleActionSummary = t.TypeOf<typeof RuleActionSummary>;
+export const RuleActionSummary = t.boolean;
+
+export type RuleActionNotifyWhen = t.TypeOf<typeof RuleActionNotifyWhen>;
+export const RuleActionNotifyWhen = t.union([
+  t.literal('onActionGroupChange'),
+  t.literal('onActiveAlert'),
+  t.literal('onThrottleInterval'),
+]);
+
+export type RuleActionFrequency = t.TypeOf<typeof RuleActionFrequency>;
+export const RuleActionFrequency = t.type({
+  summary: RuleActionSummary,
+  notifyWhen: RuleActionNotifyWhen,
+  throttle: t.union([RuleActionThrottle, t.null]),
```

Comment:
Why do we need to return `null` when `throttle` is not set while it could be an optional field resolved to `undefined` if there is no a value. This was it won't be serialised and sent to the client. On top of that we try to avoid to use `null` in our API according to https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/common/detection_engine/rule_schema/model/rule_schemas.ts.

Reply:
We are not using io-ts in platform, and we want to keep using `kbn-schema`. I think we are fine with that for now

Reply:
`throttle` is set to `null` everywhere in `security solution`, `alert framework` and `triggering actions` plugins when `notifyWhen` attribute is not `onThrottleInterval`.

Our rule schemas are also use it here https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/server/lib/detection_engine/rule_schema/model/rule_schemas.ts

Even if we want to avoid setting it to `null` and just keep undefined in case it is not set, we would need to go through many places make sure everything works as expected. So, I would not do this as part of this PR.